src/kits/web/SpyRESTWeblet.sedona

```diff
@@ -0,0 +1,695 @@
+    else if (p.equals("logs")) updateLogLevel(req, res)
+    else
+    {
+      // Error: Method not supported
+    }
+  }
+
+  override void get(WebReq req, WebRes res)
+  {
+    // start header
+    res.json()
+    nav(req, res)
+
+    // route to html pages
+    Str p = ""
+    // ignore
+    if (req.path.size > 1) p = req.path.names[1]
+
+    if (p.equals("app"))        app(req, res)
+    else if (p.equals("kits"))  kits(req, res)
+    else if (p.equals("sox"))   sox(req, res)
+    else if (p.equals("echo"))  echo(req, res)
+    else if (p.equals("logs"))  logs(req, res)
+    else index(req, res)
+
+    // standard footer
+    res.jsonEnd(0)
+  }
+
+  private void nav(WebReq req, WebRes res)
+  {
+    int count = req.params.size
+    Str p = ""
+    for (int i=0; i<count; ++i)
+    {
+      Str param = req.params.names[i]
+      if(param.startsWith("Host"))
+        p = req.params.values[i]
+    }
+    if (p.length() > 0) {
+      res.jsonStrN("pathRef").w(": \"")
+      res.w("http://").w(p.trim()).w(req.uri).w("\",")
+    }
+  }
+
+////////////////////////////////////////////////////////////////
+// Index
+////////////////////////////////////////////////////////////////
+
+  public void index(WebReq req, WebRes res)
+  {
+    App app = Sys.app
+    PlatformService plat = app.platform
+    res.jsonStrN("summary").w(": {");
+
+    res.jsonStrN("platformType").w(":\"").w(plat.type.kit.name).w("::").w(plat.type.name).w("\",");
+    res.jsonKV("platformId", plat.platformId.toStr()).w(",");
+    res.jsonKV("name", app.name).w(",");
+    res.jsonStrN("uptime").w(":\"").printElapsed(Sys.bootTime).w("\",");
+    res.jsonStrN("cycleCount").w(":\"").printLong(app.cycleCount)
+    res.w("\"").jsonEnd(0)
+    res.w(",").nl()
+
+    res.jsonStrN("indexGET").w(": [")
+    res.jsonStrN("/spyREST/app").w(",")
+    res.jsonStrN("/spyREST/kits").w(",")
+    res.jsonStrN("/spyREST/logs").w(",")
+    res.jsonStrN("/spyREST/sox").w(",")
+    res.jsonStrN("/spyREST/echo").nl()
+    res.w("],").nl()
+
+    res.jsonStrN("indexPOST").w(": [")
+    res.jsonStrN("/spyREST/backup").w(",")
+    res.jsonStrN("/spyREST/logs").nl()
+    res.w("],").nl()
+
+    res.jsonStrN("weblets").w(": {")
+    int count = 0
+    for (Weblet w = WebService.weblets; w != null; w = w.nextWeblet)
+    {
+      if (count > 0) res.w(",") // don't put comma for first element
+      res.jsonKV(w.prefix(), w.description())
+      count++
+    }
+    res.jsonEnd(0)
+  }
+
+////////////////////////////////////////////////////////////////
+// App
+////////////////////////////////////////////////////////////////
+
+  void app(WebReq req, WebRes res)
+  {
+    Path path = req.path
+    if (req.path.size > 2)
+    {
+      int id = req.path.names[2].parseInt()
+      Component c = Sys.app.lookup(id)
+      if (c != null)
+      {
+        appComponent(req, res, c)
+        return
+      }
+    }
+    appComponent(req, res, Sys.app)
+  }
+
+  void appComponent(WebReq req, WebRes res, Component c)
```

Comment:
I think we shouldn't display all component object, at least for user component due to security reason. maybe we should add authentication for even GET request. It seems the SpyWeblet has this issue, too. If this kit is only for use within a trusted intranet environment, better to point this out in document and let user know the security risk.

Reply:
Agree. A usage note saying that 'Web Service' (including spy and spyREST) should not be enabled in production deployments, only during debugging. With a formal REST service, authC and authZ should be properly handled.

Reply:
yeah, thanks.